session/server.go

```diff
@@ -43,10 +47,13 @@ func (m *mailboxSession) start(session *Session,
 
 	ecdh := &keychain.PrivKeyECDH{PrivKey: session.LocalPrivateKey}
 
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
why do we not derive this from a parent context?

Reply:
because this gets spun off into a go-routine. So we want it to be long lived passed the point of the initial call

Reply:
we can eventually change things such that `start` is only ever called from a single event loop in which case then we can pass a long-lived context to it from it. 

but since this is a bit of a re-design, we can do it as a follow up